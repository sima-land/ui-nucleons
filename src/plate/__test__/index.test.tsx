import { createRef } from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Plate } from '..';

describe('Plate', () => {
  it('should renders without props', () => {
    const { container } = render(<Plate />);

    expect(container).toMatchSnapshot();
  });

  it('should handle props', () => {
    const ref = createRef<HTMLInputElement | null>();
    const spy = jest.fn();

    const { container } = render(
      <Plate ref={ref} shadow='z3' rounds='m' className='test-class' onClick={spy}>
        Test
      </Plate>,
    );

    expect(container).toMatchSnapshot();

    expect(container.querySelectorAll('div')).toHaveLength(1);
    expect(container.querySelectorAll('div.test-class')).toHaveLength(1);
    expect(container.querySelector('div.test-class')).toBe(ref.current);

    expect(container.querySelectorAll('div[data-testid="plate"]')).toHaveLength(1);
    expect(container.querySelector('div[data-testid="plate"]')).toBe(ref.current);

    expect(spy).toHaveBeenCalledTimes(0);
    fireEvent.click(container.querySelector('div') as HTMLElement);
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should handle "data-testid" prop', () => {
    const ref = createRef<HTMLInputElement | null>();
    const spy = jest.fn();

    const { container } = render(
      <Plate
        ref={ref}
        shadow='z3'
        rounds='m'
        className='test-class'
        onClick={spy}
        data-testid='test-plate'
      >
        Test
      </Plate>,
    );

    expect(container).toMatchSnapshot();

    expect(container.querySelectorAll('div')).toHaveLength(1);
    expect(container.querySelectorAll('div.test-class')).toHaveLength(1);
    expect(container.querySelector('div.test-class')).toBe(ref.current);

    expect(container.querySelectorAll('div[data-testid="test-plate"]')).toHaveLength(1);
    expect(container.querySelector('div[data-testid="test-plate"]')).toBe(ref.current);

    expect(spy).toHaveBeenCalledTimes(0);
    fireEvent.click(container.querySelector('div') as HTMLElement);
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should renders with small rounds', () => {
    const { container } = render(<Plate rounds='s'>Test</Plate>);

    expect(container).toMatchSnapshot();
  });
});
