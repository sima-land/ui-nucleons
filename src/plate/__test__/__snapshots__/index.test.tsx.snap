// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plate should handle "data-testid" prop 1`] = `
<div>
  <div
    class="root test-class shadow-z3 rounds-md__all"
    data-testid="test-plate"
  >
    Test
  </div>
</div>
`;

exports[`Plate should handle props 1`] = `
<div>
  <div
    class="root test-class shadow-z3 rounds-md__all"
    data-testid="plate"
  >
    Test
  </div>
</div>
`;

exports[`Plate should renders with small rounds 1`] = `
<div>
  <div
    class="root shadow-z1 rounds-sm__all"
    data-testid="plate"
  >
    Test
  </div>
</div>
`;

exports[`Plate should renders without props 1`] = `
<div>
  <div
    class="root shadow-z1 rounds-sm__all"
    data-testid="plate"
  />
</div>
`;
